index/scorch/optimize_knn.go

```diff
@@ -101,6 +101,11 @@ func (o *OptimizeVR) Finish() error {
 					wg.Done()
 				}()
 				for _, field := range fields {
+					// Noop if the vector field or its index data is supposed to be deleted
+					if info, ok := o.snapshot.updatedFields[field]; ok && (info.Deleted || info.Index) {
```

Comment:
As far as readability goes, I have to say this looks a bit off-putting, perhaps commentary here on why `|| info.Index` will help.